### DISCUSSION TITLE
How to understand parameter timeout?

### DISCUSSION THREAD

**Discussion by Caratpine** [projectdiscovery/nuclei#4187] [Q&A] [answered]
Nuclei version: 2.8.8

```
-timeout int               time to wait in seconds before timeout (default 10)
```
Does timeout refer to the timeout of each request, or does it refer to the time for the entire nuclei command to run?
When I set timeout = 5, the entire nuclei command took more than 5 seconds to run, so I guess it doesn't refer to the command running time, right?

**Comment by ehsandeep** [comment, marked as answer]
@Caratpine it refers to each request, also will suggest updating nuclei to the latest version and try doing the same.

**Reply by Caratpine** [reply]
Hi, @ehsandeep thanks for your reply, and I am currently unable to upgrade to the latest version because the syntax of the Nuclei template has changed and we are still using the old template.
